src/Features/Core/Portable/SignatureHelp/SignatureContext.cs

```diff
@@ -0,0 +1,148 @@
+        /// </summary>
+        public Document Document { get; }
+
+        /// <summary>
+        /// The caret position when Signature Help was triggered.
+        /// </summary>
+        public int Position { get; }
+
+        /// <summary>
+        /// The triggering action that caused Signature Help to be started.
+        /// </summary>
+        public SignatureHelpTrigger Trigger { get; }
+
+        /// <summary>
+        /// The options that Signature Help was started with.
+        /// </summary>
+        public OptionSet Options { get; }
+
+        /// <summary>
+        /// The cancellation token to use for this operation.
+        /// </summary>
+        public CancellationToken CancellationToken { get; }
+
+        public TextSpan ApplicableSpan => _applicableSpan;
+
+        public SignatureHelpState State => _state;
+
+        /// <summary>
+        /// Creates a <see cref="SignatureContext"/> instance.
+        /// </summary>
+        public SignatureContext(
+            Document document,
+            int position,
+            SignatureHelpTrigger trigger,
+            OptionSet options,
+            CancellationToken cancellationToken)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            this.Document = document;
+            this.Position = position;
+            this.Trigger = trigger;
+            this.Options = options;
+            this.CancellationToken = cancellationToken;
+            _items = new List<SignatureHelpItem>();
+        }
+
+        public void AddItem(SignatureHelpItem item)
+        {
+            if (item == null)
+            {
+                throw new ArgumentNullException(nameof(item));
+            }
+
+            _items.Add(item);
+        }
+
+        public void AddItems(IEnumerable<SignatureHelpItem> items)
+        {
+            if (items == null)
+            {
+                throw new ArgumentNullException(nameof(items));
+            }
+
+            foreach (var item in items)
+            {
+                AddItem(item);
+            }
+        }
+
+        public void SetApplicableSpan(TextSpan span)
+        {
+            _applicableSpan = span;
+        }
+
+        public void SetState(SignatureHelpState state)
+        {
+            _state = state;
+        }
+
+        internal SignatureList ToSignatureList(SignatureHelpProvider provider)
+        {
+            if (_items == null || !_items.Any() || _state == null)
+            {
+                return null;
+            }
+
+            var items = Filter(_items, _state.ArgumentNames);
+            return new SignatureList(provider, items.ToList(), _applicableSpan, _state.ArgumentIndex, _state.ArgumentCount, _state.ArgumentName);
```

Comment:
It was left over from merging with some of your changes here when I moved this PR forward. Good catch!